lib/workspace.rb

```diff
@@ -0,0 +1,58 @@
+# workspace file for keeping all things slack
+require "dotenv"
+require "httparty"
+
+require_relative "user"
+require_relative "channels"
+
+Dotenv.load
+
+module Slack
+  class Workspace
+    URL = "https://slack.com/api/users.list"
+    attr_reader :users, :channels
+    attr_accessor :selected
+
+    def initialize
+      @users = User.list
+      @channels = Channel.list
+      @selected = nil
+    end
+
+    def select_user(user_descriptor)
+      users.find do |user|
+        if user_descriptor == user.name || user_descriptor == user.slack_id
+          @selected = user
+        end
+      end
+
+      if @selected == nil
```

Comment:
There's a subtle error in the way you have it written.  Consider:

1.  I select a valid user
2.  Then I try to select an invalid user name
3.  Since `@selected` is **not** `nil` we won't get the returned error message.